core/src/main/java/io/grpc/Metadata.java

```diff
@@ -177,6 +178,14 @@ public T apply(MetadataEntry entry) {
   }
 
   /**
+   * Returns immutable representation of metadata store
+   * @return ImmutableMap copy of internal store
+   */
+  public ImmutableMap<String, List<MetadataEntry>> getStore() {
```

Comment:
MetadataEntry is a private class, you won't be able to access it at all.

Also, ImmutableMap is too specific of a return value.  It should be a Map in any case.